src/rgw/rgw_lua_utils.cc

```diff
@@ -41,4 +42,23 @@ void stack_dump(lua_State* L) {
   std::cout << "--------------- Stack Dump Finished ---------------" << std::endl;
 }
 
+void set_package_path(lua_State* L, const std::string& install_dir) {
+  if (install_dir.empty()) {
+    return;
+  }
+  const auto path = install_dir+"/share/lua/"+CEPH_LUA_VERSION+"/?.lua";  
+  lua_getglobal(L, "package");
+  if (!lua_istable(L, -1)) {
+    return;
+  }
+  lua_pushstring(L, "path");
+  lua_pushstring(L, path.c_str());
+  lua_settable(L, -3); 
+  
+  const auto cpath = install_dir+"/lib/lua/"+CEPH_LUA_VERSION+"/?.so";
+  lua_pushstring(L, "cpath");
+  lua_pushstring(L, cpath.c_str());
+  lua_settable(L, -3); 
```

Comment:
```suggestion
  lua_getglobal(L, "package");
  if (!lua_istable(L, -1)) {
    return;
  }
  pushstring(L, path);
  lua_setfield(L, -2, "path"); 
  
  const auto cpath = install_dir+"/lib/lua/"+CEPH_LUA_VERSION+"/?.so";
  pushstring(L, cpath);
  lua_setfield(L, -2, "cpath"); 
```

Reply:
This also overrides the "default" search paths. Do you want that?

Reply:
Also: if the luarocks installation has other packages installed, your allowlist is circumvented. Should the rgw path always be some transient sandbox? What if it's the system directory/default?

Reply:
> This also overrides the "default" search paths. Do you want that?

yes, i want the rgw lua search path to be completely detached from whatever is configured and installed on the machine 

> Also: if the luarocks installation has other packages installed, your allowlist is circumvented. 

why? i don't search in the regular path

> Should the rgw path always be some transient sandbox? What if it's the system directory/default?

yes, by default it is set to `/tmp/luarocks`. note that i delete it before i install new packages to it (to more easily handle package deletions). if this directory is configured to the systems luarocks directory, i will mess up the packages on the machine. but this will just be misconfiguration

Reply:
> > This also overrides the "default" search paths. Do you want that?
> 
> yes, i want the rgw lua search path to be completely detached from whatever is configured and installed on the machine

K

> > Also: if the luarocks installation has other packages installed, your allowlist is circumvented.
> 
> why? i don't search in the regular path

Ok

> > Should the rgw path always be some transient sandbox? What if it's the system directory/default?
> 
> yes, by default it is set to `/tmp/luarocks`. note that i delete it before i install new packages to it (to more easily handle package deletions). if this directory is configured to the systems luarocks directory, i will mess up the packages on the machine. but this will just be misconfiguration

Perhaps you should catch that misconfiguration or at least document it.

Reply:
> Perhaps you should catch that misconfiguration or at least document it.

will document as part of this parameter's documentation.